dev/src/bulk-writer.ts

```diff
@@ -740,27 +914,69 @@ export class BulkWriter {
    *
    * @private
    */
-  private sendBatch(batch: BulkCommitBatch): void {
+  private sendBatch(
+    batch: BulkCommitBatch,
+    isRetryBatch = false
+  ): Promise<void> {
     const success = this.rateLimiter.tryMakeRequest(batch.opCount);
     assert(success, 'Batch should be under rate limit to be sent.');
-    batch.bulkCommit().then(() => {
+    return batch.bulkCommit().then(() => {
       // Remove the batch from the BatchQueue after it has been processed.
-      const batchIndex = this.batchQueue.indexOf(batch);
+      const batchQueue = isRetryBatch
+        ? this._retryBatchQueue
+        : this._batchQueue;
+      const batchIndex = batchQueue.indexOf(batch);
       assert(batchIndex !== -1, 'The batch should be in the BatchQueue');
-      this.batchQueue.splice(batchIndex, 1);
+      batchQueue.splice(batchIndex, 1);
 
       this.sendReadyBatches();
     });
   }
 
+  private wrapRetry(
+    promise: Promise<WriteResult>,
+    documentRef: firestore.DocumentReference,
+    retryCount: number,
+    operationType: 'create' | 'set' | 'update' | 'delete',
+    retryFn: () => Promise<WriteResult>
```

Comment:
This is the original function isn't it? Could we just call it in this method ourselves, which would drop the `promise` argument.